db/snapshot/backup_task.cc

```diff
@@ -99,52 +111,132 @@ future<> backup_task_impl::upload_component(sstring name) {
     }
 }
 
-future<> backup_task_impl::do_backup() {
-    if (!co_await file_exists(_snapshot_dir.native())) {
-        throw std::invalid_argument(fmt::format("snapshot does not exist at {}", _snapshot_dir.native()));
+namespace {
+[[maybe_unused]] std::vector<std::tuple<size_t, std::vector<sstring>>> greedy_partitioning(std::vector<std::tuple<size_t, sstring>>&& files,
+                                                                                           size_t partitions) {
+    std::vector<std::tuple<size_t, std::vector<sstring>>> result(partitions);
+    std::ranges::sort(files, [](const auto& lhs, const auto& rhs) { return std::get<0>(lhs) > std::get<0>(rhs); });
```

Comment:
Use std::greater<>()

Reply:
Ah, you also have a projection. I would make the tuple a struct and use &whatever_meaningful_name_you_choose_instead_of_a_meaningless_tuple::data_size as the projection.

Reply:
Hopefully, this makes it more readable and maintainable.